README
NAME
    Alien::SNMP::MIBDEV - Alien package for the Net-SNMP library

VERSION
    Version 2.000000

SYNOPSIS
     use Alien::SNMP::MIBDEV;
     # then it's just like SNMP.pm
 
     say Alien::SNMP::MIBDEV->bin_dir;
     # where the net-snmp apps (snmptranslate, etc) live

DESCRIPTION
    Alien::SNMP::MIBDEV is mainly used for netdisco-mibs development where
    standard settings do not suffice. It's not intended for other purposes.

    Alien::SNMP::MIBDEV downloads and installs the Net-SNMP 5.8 library and
    associated perl modules.

    This is based on Alien::SNMP::MAXTC.

    Compared to the standard module MAX_IMPORTS has been raised to 512.

    The library is built with the following options:

    MAX_IMPORTS set to 512
    "--with-pic"
    "--disable-agent"
    "--disable-manuals"
    "--disable-scripts"
    "--disable-mibs"
    "--enable-ipv6"
    "--with-mibs="""
    "--with-perl-modules"
    "--disable-embedded-perl"
    "--with-defaults"

METHODS
  bin_dir
     my $bin_dir = Alien::SNMP::MIBDEV->bin_dir;

    Returns the location of the net-snmp apps (snmptranslate, etc).

  cflags
     my $cflags = Alien::SNMP::MIBDEV->cflags;

    Returns the C compiler flags.

  libs
     my $libs = Alien::SNMP::MIBDEV->libs;

    Returns the linker flags.

SEE ALSO
    Alien
    Alien::Base
    Alien::SNMP
    Alien::SNMP::MAXTC
    SNMP
        The Perl5 'SNMP' Extension Module for the Net-SNMP SNMP package.
        Depends on libnetsnmp and the corresponding version is installed
        along with the C library.

AUTHOR
    Eric A. Miller, "<emiller at cpan.org>"

    Oliver Gorwits

    Nick Nauwelaerts

LICENSE AND COPYRIGHT
    Copyright 2015 Eric A. Miller.

    Copyright 2020 Oliver Gorwits.

    Copyright 2021 Nick Nauwelaerts.

    This program is distributed under the (Revised) BSD License:
    <http://www.opensource.org/licenses/BSD-3-Clause>

    Redistribution and use in source and binary forms, with or without
    modification, are permitted provided that the following conditions are
    met:

    * Redistributions of source code must retain the above copyright notice,
    this list of conditions and the following disclaimer.

    * Redistributions in binary form must reproduce the above copyright
    notice, this list of conditions and the following disclaimer in the
    documentation and/or other materials provided with the distribution.

    * Neither the name of Eric A. Miller's Organization nor the names of its
    contributors may be used to endorse or promote products derived from
    this software without specific prior written permission.

    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
    IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
    TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
    PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER
    OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
    EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
    PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
    PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
    LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
    NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
    SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.



Changes
2.040000 - xxx

  * migrate to alienfile

2.020000 - 2021-07-08

  * drop test for snmp.pm module for now

2.000000 - 2021-06-17

  * rework Alien::SNMP::MAXTC in Alien::SNMP::MIBDEV
  * move to net::snmp 5.8 as base
  * drop MAXTC patching
  * patch MAX_IMPORTS to 512
  * general cleanup

1.001000 - 2020-05-01

  * Fix Alien dependencies spec (thibaultduponchelle)

1.000001 - 2017-08-12

  * Merged master

1.000000 - 2017-08-12

  * CPAN release

0.030000 - 2017-04-14

  * Move net-snmp upstream to another repo

0.020000 - 2017-03-06

  * Create Alien::SNMP::MAXTC

0.010000 - 2015-10-16

  * First version, initial release


Makefile.PL
use strict;
use warnings;
use ExtUtils::MakeMaker;
use Alien::Build::MM;

my $abmm = Alien::Build::MM->new;

WriteMakefile($abmm->mm_args(
#	SIGN         => 1,
	ABSTRACT     => 'Alien wrapper for Net-SNMP',
	DISTNAME     => 'Alien-SNMP-MIBDEV',
	NAME         => 'Alien::SNMP::MIBDEV',
	VERSION_FROM => 'lib/Alien/SNMP/MIBDEV.pm',
	LICENSE      => 'bsd',
        AUTHOR       => [ 'Eric A. Miller <emiller AT cpan DOT org>',
                          'Oliver Gorwits <oliver AT cpan DOT org>',
                          'Nick Nauwelaerts <inphobia AT cpan DOT org>',
                        ],
	META_MERGE => {
		resources => {
			homepage => 'https://github.com/inphobia/Alien-SNMP-MIBDEV',
			bugtracker => 'https://github.com/inphobia/Alien-SNMP-MIBDEV/issues',
			repository  => 'https://github.com/inphobia/Alien-SNMP-MIBDEV',
			mailinglist => 'https://lists.sourceforge.net/lists/listinfo/snmp-info-users',
			irc => 'irc://irc.libera.chat/#netdisco',
		},
	},
));

sub MY::postamble {
  $abmm->mm_postamble(@_);
}

sub MY::install {
  $abmm->mm_install(@_);
}


lib/Alien/SNMP/MAXTC.pm
package Alien::SNMP::MAXTC;

use strict;
use warnings;
use 5.010001;
use parent qw(Alien::Base);

our $VERSION = '1.002000';

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Alien::SNMP::MAXTC - Alien package for the Net-SNMP library

=head1 VERSION

Version 1.000001

=cut

=head1 SYNOPSIS

 use Alien::SNMP::MAXTC;
 # then it's just like SNMP.pm
 
 say Alien::SNMP::MAXTC->bin_dir;
 # where the net-snmp apps (snmptranslate, etc) live

=head1 DESCRIPTION

L<Alien::SNMP::MAXTC> downloads and installs the Net-SNMP library and
associated perl modules.

This is a fork of L<Alien::SNMP> with some changed options. We may merge-back
one day.

The library is built with the following options:

=over

=item MAXTC set to 16k

=item C<--disable-agent>

=item C<--disable-manuals>

=item C<--disable-scripts>

=item C<--disable-mibs>

=item C<--enable-ipv6>

=item C<--with-mibs="">

=item C<--with-perl-modules>

=item C<--disable-embedded-perl>

=item C<--with-defaults>

=back

=head1 METHODS

=head2 bin_dir

 my $bin_dir = Alien::SNMP::MAXTC->bin_dir;

Returns the location of the net-snmp apps (snmptranslate, etc).

=head2 cflags

 my $cflags = Alien::SNMP::MAXTC->cflags;

Returns the C compiler flags.

=head2 libs

 my $libs = Alien::SNMP::MAXTC->libs;

Returns the linker flags.

=head1 SEE ALSO

=over 4

=item L<Alien>

=item L<Alien::Base>

=item L<SNMP>

The Perl5 'SNMP' Extension Module for the Net-SNMP SNMP package.  Depends on
libnetsnmp and the corresponding version is installed along with the C
library.

=back

=head1 AUTHOR

Eric A. Miller, C<< <emiller at cpan.org> >>

=head1 LICENSE AND COPYRIGHT

Copyright 2015 Eric A. Miller.

This program is distributed under the (Revised) BSD License:
L<http://www.opensource.org/licenses/BSD-3-Clause>

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions
are met:

* Redistributions of source code must retain the above copyright
notice, this list of conditions and the following disclaimer.

* Redistributions in binary form must reproduce the above copyright
notice, this list of conditions and the following disclaimer in the
documentation and/or other materials provided with the distribution.

* Neither the name of Eric A. Miller's Organization
nor the names of its contributors may be used to endorse or promote
products derived from this software without specific prior written
permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

=cut


lib/Alien/SNMP/MIBDEV.pm
package Alien::SNMP::MIBDEV;

use strict;
use warnings;
use 5.010001;
use parent qw(Alien::Base);

our $VERSION = '2.010000'; # TRIAL

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

Alien::SNMP::MIBDEV - Alien package for the Net-SNMP library

=head1 VERSION

Version 2.000000

=cut

=head1 SYNOPSIS

 use Alien::SNMP::MIBDEV;
 # then it's just like SNMP.pm
 
 say Alien::SNMP::MIBDEV->bin_dir;
 # where the net-snmp apps (snmptranslate, etc) live

=head1 DESCRIPTION

L<Alien::SNMP::MIBDEV> is mainly used for netdisco-mibs development where
standard settings do not suffice. It's not intended for other purposes.

L<Alien::SNMP::MIBDEV> downloads and installs the Net-SNMP 5.8 library and
associated perl modules.

This is based on L<Alien::SNMP::MAXTC>.

Compared to the standard module MAX_IMPORTS has been raised to 512.

The library is built with the following options:

=over

=item MAX_IMPORTS set to 512

=item C<--with-pic>

=item C<--disable-agent>

=item C<--disable-manuals>

=item C<--disable-scripts>

=item C<--disable-mibs>

=item C<--enable-ipv6>

=item C<--with-mibs="">

=item C<--with-perl-modules>

=item C<--disable-embedded-perl>

=item C<--with-defaults>

=back

=head1 METHODS

=head2 bin_dir

 my $bin_dir = Alien::SNMP::MIBDEV->bin_dir;

Returns the location of the net-snmp apps (snmptranslate, etc).

=head2 cflags

 my $cflags = Alien::SNMP::MIBDEV->cflags;

Returns the C compiler flags.

=head2 libs

 my $libs = Alien::SNMP::MIBDEV->libs;

Returns the linker flags.

=head1 SEE ALSO

=over 4

=item L<Alien>

=item L<Alien::Base>

=item L<Alien::SNMP>

=item L<Alien::SNMP::MAXTC>

=item L<SNMP>

The Perl5 'SNMP' Extension Module for the Net-SNMP SNMP package.  Depends on
libnetsnmp and the corresponding version is installed along with the C
library.

=back

=head1 AUTHOR

Eric A. Miller, C<< <emiller at cpan.org> >>

Oliver Gorwits

Nick Nauwelaerts

=head1 LICENSE AND COPYRIGHT

Copyright 2015 Eric A. Miller.

Copyright 2020 Oliver Gorwits.

Copyright 2021 Nick Nauwelaerts.

This program is distributed under the (Revised) BSD License:
L<http://www.opensource.org/licenses/BSD-3-Clause>

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions
are met:

* Redistributions of source code must retain the above copyright
notice, this list of conditions and the following disclaimer.

* Redistributions in binary form must reproduce the above copyright
notice, this list of conditions and the following disclaimer in the
documentation and/or other materials provided with the distribution.

* Neither the name of Eric A. Miller's Organization
nor the names of its contributors may be used to endorse or promote
products derived from this software without specific prior written
permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

=cut


Build.PL
#!/usr/bin/env perl

use strict;
use warnings;
use 5.010001;
use utf8;

use lib 'inc';
use My::ModuleBuild;

my $conf_cmd = [
  '%c',                     '--with-pic',
  '--prefix=%s',            '--disable-agent',
  '--disable-scripts',      '--disable-mibs',
  '--enable-ipv6',          '--with-mibs=""',
  '--with-perl-modules',    '--disable-embedded-perl',
  '--disable-manuals',      '--with-defaults',
];

my %module_build_args = (
  alien_autoconf_with_pic => 0,
  alien_build_commands =>
    ["%x -I../../inc -MMy::AlienPatch -e alien_patch", $conf_cmd, 'make'],

  # Not using FFI, default_store library only dynamic and needed for XS
  alien_isolate_dynamic => 0,
  alien_name            => 'netsnmp',

  # Headers referenced as net-snmp/xxx.h, autodiscovery appends net-snmp
  alien_provides_cflags => '-I${pcfiledir}/include',
  alien_provides_libs   => '-lnetsnmp',
  alien_repository      => {
    protocol       => 'https',
    host           => 'github.com',
    location       => '/netdisco/upstream-sources/blob/master/net-snmp/',
    exact_filename => 'net-snmp-5.8.tar.gz?raw=true',
  },
  alien_stage_install => 0,
  build_requires      => {
    "Alien::Base"        => '0.020',
    "Software::License" => '0',
  },
  configure_requires  => {
    "perl"               => '5.010001',
    "Alien::Base::ModuleBuild" => '0',
    "File::ShareDir"     => '1.03',
    "Module::Build"      => '0.42',
    "PkgConfig"          => '0.14026',
    "Test2::Suite"       => '0.000067'
  },
  create_license    => 1,
  create_readme     => 1,
  dist_abstract     => 'Build and install Net-SNMP',
  dist_author       => [
    'Eric A. Miller <emiller AT cpan DOT org>',
    'Oliver Gorwits <oliver AT cpan DOT org>',
    'Nick Nauwelaerts <inphobia AT cpan DOT org>',
  ],
  dist_name         => 'Alien-SNMP-MIBDEV',
  dist_version_from => 'lib/Alien/SNMP/MIBDEV.pm',
  license           => 'bsd',
  module_name       => "Alien::SNMP::MIBDEV",
  release_status    => 'testing',
  recommends        => {},
  requires          => {
    "Alien::Base"    => '0.020',
    "File::ShareDir" => '1.03',
    "parent"         => '0',
    "perl"           => '5.010001',
  },
  test_requires => {
    "Test::Alien" => 0,
    "Test::More" => '0.94',
    "Test::Manifest" => '0.9',
    "perl" => '5.010001'
  },
  meta_merge => {
    resources => {
      repository => 'https://github.com/inphobia/Alien-SNMP-MIBDEV',
      bugtracker => 'https://github.com/inphobia/Alien-SNMP-MIBDEV/issues',
      MailingList => 'https://lists.sourceforge.net/lists/listinfo/snmp-info-users',
      IRC => 'irc://irc.libera.chat/#netdisco',
    },
  },
);

my $builder = My::ModuleBuild->new(%module_build_args);

$builder->create_build_script;
